### PR TITLE
feat: Forbid expression without alias in mview definition

### LINKED ISSUE #2404: Forbid aggregation without alias in mview definition
It's implemented in the legacy frontend.

Consider the following SQL:

```sql
create table t1 (v1 int);
create materialized view mv1 as select sum(v1), count(v1) from t1;
# How to select from mv1?
# wildcard is ok
select * from mv1;
# But we cannot project it by a name.
# sum(v1) is definitely wrong
select sum(v1) from mv1;
# A good practice
create materialized view mv1 as select sum(v1) as sum_v1, count(v1) as count_v1 from t1;
select sum_v1 from mv1;
```

### PR DESCRIPTION
**Pull request by xxchan**
## What's changed and what's your intention?
as title

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
close #2404

### CODE REVIEW

**Comment by st1page** [comment]
maybe we can add some bind error test in planner test

**Comment by st1page** [comment]
> > maybe we can add some bind error test in planner test
> 
> need to update test runner since its binder error is only for queries. Also the error is actually outside binder. I am wondering whether it is proper 😇

oh, I get it. so we can improve our planner test framework later.

File: src/frontend/src/handler/create_mv.rs [hunk 1]
```diff
@@ -45,6 +47,22 @@ pub fn gen_create_mv_plan(
...
+        if select
+            .select_items
+            .iter()
+            .zip_eq(select.aliases.iter())
+            .any(|(select_item, alias)| {
+                matches!(select_item, ExprImpl::AggCall(_)) && alias.is_none()
+            })
+        {
+            return Err(ErrorCode::BindError(
+                "An alias must be specified for an aggregation function".to_string(),
```
Comment (TennyZhuang) [inline src/frontend/src/handler/create_mv.rs:60]: IMO, it should also apply to normal function.

Comment (TennyZhuang) [reply src/frontend/src/handler/create_mv.rs:60]: But not necessary in this PR

**Review by TennyZhuang — APPROVED** [review]
LGTM

**Review by xiangjinwu — APPROVED** [review]

File: src/frontend/src/handler/create_mv.rs (hunk 1 above)
Comment (xiangjinwu) [reply src/frontend/src/handler/create_mv.rs:60]: And also literals... Actually the only thing to allow without explicit alias is `InputRef`, which is already getting implicit alias assigned in bind_project. Maybe we can simply check that all `aliases` in `BoundSelect` is not `None` here.

Comment (xxchan) [reply src/frontend/src/handler/create_mv.rs]: Got it & updated. I misunderstood the `alias` in `BoundSelect`, and thought it's only user's inputs.

**Review by xiangjinwu — APPROVED** [review]

**Pull request merged**